src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java

```diff
@@ -109,36 +98,25 @@ public JiraSendInfoResponse sendBuildInfoToJiraSite(
             return JiraBuildInfoResponse.skippedIssueKeysNotFound(siteConfig.getSite());
         }
 
-        final Optional<String> maybeCloudId = getCloudIdFor(resolvedSiteConfig);
+        final Optional<String> maybeCloudId = getCloudIdFor(jiraSite);
 
         if (!maybeCloudId.isPresent()) {
-            return JiraCommonResponse.failureSiteNotFound(resolvedSiteConfig);
-        }
-
-        final Optional<String> maybeAccessToken = getAccessTokenFor(siteConfig, maybeSecret.get());
-
-        if (!maybeAccessToken.isPresent()) {
-            return JiraCommonResponse.failureAccessToken(resolvedSiteConfig);
+            return JiraCommonResponse.failureSiteNotFound(jiraSite);
         }
 
         final Builds buildInfo = createJiraBuildInfo(request, issueKeys);
 
-        final PostUpdateResult<BuildApiResponse> postUpdateResult =
-                sendBuildInfo(
-                        maybeCloudId.get(), maybeAccessToken.get(), resolvedSiteConfig, buildInfo);
-
-        if (postUpdateResult.getResponseEntity().isPresent()) {
+        try {
             return handleBuildApiResponse(
-                    resolvedSiteConfig, postUpdateResult.getResponseEntity().get());
-        } else {
-            final String errorMessage = postUpdateResult.getErrorMessage().orElse("");
-            return handleBuildApiError(resolvedSiteConfig, errorMessage);
+                    jiraSite, buildsApi.sendBuild(siteConfig.getWebhookUrl(), buildInfo));
+        } catch (ApiUpdateFailedException e) {
+            return handleBuildApiError(jiraSite, e.getMessage());
```

Comment:
The error handling changed a bit here. The `sendBuild()` method now throws an `ApiUpdateFailedException` in case of an error instead of returning an object with an error message (and leaking specifics of the OkHttp library in the process).